astlib/compmisc.mli

```diff
@@ -0,0 +1 @@
+val read_clflags_from_env : unit -> unit
```

Comment:
It feels like this shouldn't be exposed in the API of Astlib. What about adding a function `Astlib.init` that calls this under the hood?

Reply:
I agree that `Astlib.Compmisc.read_clflags_from_env` is a bad idea. Why do you think `Astlib.init` would be better than `Astlib.read_clflags_from_env`? The latter would give a hint about what the function does. Do you prefer the former in order to keep open what it does so that it's behavior can be changed in the future?

Reply:
> Do you prefer the former in order to keep open what it does so that it's behavior can be changed in the future?

That's what I was thinking. Since Astlib is a library, it's not clear why it cares about reading command line flags from the environment.

And to be fair, it would improve reproducibility if we didn't have to do that, but that's another story.

Reply:
Yes, needing that functionality in Astlib is really a bit unfortunate.

I've called it `Astlib.init` now as you've suggested. Do you think it would be good to add a brief documentation about it saying what it does? Or would documentation get in the way of hiding the behavior and leaving an open door to change it? The reason why I would add it is that without documentation, from an Astlib user's point of view, `init` might be a bit confusing: why does a library need to be initialized? Is it always necessary/advisable to run `Astlib.init` to use `Astlib`? The answer should be "no", but without context the user wouldn't know that.

Also, is removing `read_clflags_from_env` from the ppxlib driver something we could consider?

Reply:
I like the idea of giving it the more explicit name `init_error_reporting_style_using_env_vars`. I've done that now.

> We could change Dune to pass a command line flag to the ppx driver instead.

Are you sure the ppxlib driver provides those command line flags? I've had a look and couldn't find them. There's a flag called `-no-color`, but that only seems to impact the printing of the diffs that are printed when there are mismatches.
Also, looking at the ppx-universe: there are over 5 downstream users that set `OCAML_COLOR` and/or `OCAML_ERROR_STYLE` in their tests (so probably there are also quite some end users who do the same). So removing this would have more impact than I expected.

Reply:
> Are you sure the ppxlib driver provides those command line flags?

It doesn't. These would need to be added as well.

> Also, looking at the ppx-universe: there are over 5 downstream users that set OCAML_COLOR and/or OCAML_ERROR_STYLE in their tests (so probably there are also quite some end users who do the same). So removing this would have more impact than I expected.

Indeed.